src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1530 @@
+                f"{tuple(self._output_array_shape)}"
+            )
+
+        log.info(f"Driz parameter kernel: {self.kernel}")
+        log.info(f"Driz parameter pixfrac: {self.pixfrac}")
+        log.info(f"Driz parameter fillval: {self.fillval}")
+        log.info(f"Driz parameter weight_type: {self.weight_type}")
+        log.debug(
+            f"Output mosaic size (nx, ny): {self._output_wcs.pixel_shape}"
+        )
+
+        # set up an empty (don't allocate arrays at this time) output model:
+        if self._output_model is None:
+            self._output_model = self.create_output_model()
+
+        self.reset_arrays(reset_output=False, n_input_models=n_input_models)
+
+    @classmethod
+    def output_model_attributes(cls, accumulate, enable_ctx, enable_var,
+                                compute_err):
+        """
+        Returns a set of string keywords that must be present in an
+        'output_model' that is provided as input at the class initialization.
+
+        Parameters
+        ----------
+
+        accumulate : bool, optional
+            Indicates whether resampled models should be added to the
+            provided ``output_model`` data or if new arrays should be
+            created.
+
+        enable_ctx : bool, optional
+            Indicates whether to create a context image. If ``disable_ctx``
+            is set to `True`, parameters ``out_ctx``, ``begin_ctx_id``, and
+            ``max_ctx_id`` will be ignored.
+
+        enable_var : bool, optional
+            Indicates whether to resample variance arrays.
+
+        compute_err : {"from_var", "driz_err"}, None, optional
+            - ``"from_var"``: compute output model's error array from
+              all (Poisson, flat, readout) resampled variance arrays.
+              Setting ``compute_err`` to ``"from_var"`` will assume
+              ``enable_var`` was set to `True` regardless of actual
+              value of the parameter ``enable_var``.
+            - ``"driz_err"``: compute output model's error array by drizzling
+              together all input models' error arrays.
+
+            Error array will be assigned to ``'err'`` key of the output model.
+
+            .. note::
+                At this time, output error array is not equivalent to
+                error propagation results.
+
+        """
+        # always required:
+        attributes = {
+            "data",
+            "wcs",
+            "wht",
+        }
+
+        if enable_ctx:
+            attributes.add("con")
+        if compute_err:
+            attributes.add("err")
+        if enable_var:
+            attributes.update(
+                ["var_rnoise", "var_poisson", "var_flat"]
+            )
+            # TODO: if we want to support adding more data to
+            # existing output models, we need to also store weights
+            # for variance arrays:
+            # var_rnoise_weight
+            # var_flat_weight
+            # var_poisson_weight
```

Comment:
is there any plan to do this in the near future?  If not, I suggest removing the comment